tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -6421,8 +6466,9 @@ bool HLSLExternalSource::MatchArguments(
       }
       pNewType = objectElement;
     }
-    else if (pArgument->uLegalComponentTypes == LICOMPTYPE_TEXTURE2D
-      || pArgument->uLegalComponentTypes == LICOMPTYPE_TEXTURE2DARRAY) {
+    else if (i != 0 && Template[pArgument->uTemplateId] == AR_TOBJ_OBJECT) {
```

Comment:
Just a potentially useful historical note: This was originally written for sampler feedback, which were the only previous built-in functions that took objects. They all have the same solution, so this just broadens it out.